pytest_factoryboy/fixture.py

```diff
@@ -42,13 +43,17 @@ def __call__(self, request: FixtureRequest) -> Any:
         return self.function(request)
 
 
-def register(factory_class: F, _name: str | None = None, **kwargs: Any) -> F:
+def register(factory_class: F | None = None, _name: str | None = None, **kwargs: Any) -> F:
```

Comment:
The type annotation is still not right for the return value, as it may return a Callable now.

Maybe we should use the typing `@overload` way

Reply:
oh, I forgot to update the annotation for return type :( 
but `overload` seems like a good match for this case